examples/tcp_reverse/src/reverse_protocol.erl

```diff
@@ -69,7 +70,11 @@ code_change(_OldVsn, StateName, StateData, _Extra) ->
 
 %% Internal.
 
-reverse_binary(B) when is_binary(B) ->
-	[list_to_binary(lists:reverse(binary_to_list(
-		binary:part(B, {0, byte_size(B)-2})
-	))), "\r\n"].
+reverse_binary(B0) when is_binary(B0) ->
+	Size = bit_size(B0),
+	<<B1:Size/integer-little>> = B0,
+	case <<B1:Size/integer-big>> of
+		<<$\n, $\r, B2/binary>> -> <<B2/binary, $\r, $\n>>;
+		<<$\n, B2/binary>> -> <<B2/binary, $\n>>;
+		<<$\r, B2/binary>> -> <<B2/binary, $\r>>
+	end.
```

Comment:
I don't think we need to change this, neither of them are really obvious and we don't care about performance.

Reply:
The idea was to return the same EOLs that the client sent. But ok :) Tomorrow.

Reply:
We definitely don't care about that in an example. :-)

Reply:
Okaaay, I'll revert that part then.

But just for the record and to get it off my chest:
```shell
$ nc localhost 5555
test
set
```
(in absence of the `-C` switch for `nc`)

Reply:
I have not seen that personally. OK feel free to fix but make sure there is a comment explaining why it's done this way.

Reply:
In a nutshell: other than `telnet` apparently does, `nc` sends a bare LF as EOL unless `-C` is given, only then will it send CRLF. `reverse_protocol` invariably removes two bytes from the end of the line.

Reply:
Comments in last commit, is that enough?

Reply:
I don't know if we need to care about CR at all. We don't support any of those platforms and they're all old or irrelevant.

Reply:
Oh well, doesn't hurt to have it anyway.